source/ncconvert/converters/GeometryConverter.cpp

```diff
@@ -329,9 +332,34 @@ class GeometryConverter::impl
             };
         }
 
-        auto ImportMesh(const std::filesystem::path& path) -> asset::Mesh
+        auto ImportMesh(const std::filesystem::path& path, const std::optional<std::string>& internalName) -> asset::Mesh
         {
             const auto scene = ::ReadFbx(path, &m_importer, meshFlags);
+
+            aiMesh* mesh = nullptr;
+
+            if (internalName.has_value())
+            {
+                for (auto* sceneMesh : std::span(scene->mMeshes, scene->mNumMeshes))
+                {
+                    if (std::string{sceneMesh->mName.C_Str()} == internalName)
+                    {
+                        mesh = sceneMesh;
+                        break;
+                    }
+                }
+                if (mesh == nullptr) throw nc::NcError("An internal mesh name was provided but no mesh was found by that name: {}. No asset will be created.", internalName.value());
+            }
+            else 
+            {
+                mesh = scene->mMeshes[0];
+            }
+
+            if (mesh->mNumVertices == 0)
+            {
+                throw nc::NcError("No vertices found for: ", path.string());
+            }
+
             auto convertedVertices = ::ConvertToMeshVertices(scene->mMeshes[0]);
```

Comment:
Here and in the return statement we're still using `scene->mMeshes[0]` unconditionally, never `mesh`.

Reply:
Great catch. Adding a unit test to address this